packages/fuels/tests/types/evm_address/src/main.sw

```diff
@@ -0,0 +1,25 @@
+contract;
+
+use std::vm::evm::evm_address::EvmAddress;
+
+abi EvmTest {
+    fn evm_address_as_input(evm_addr: EvmAddress) -> bool;
+    fn evm_address_from_literal() -> EvmAddress;
+    fn evm_address_from_argument(raw_address: b256) -> EvmAddress;
+}
+
+impl EvmTest for Contract {
+    fn evm_address_as_input(evm_addr: EvmAddress) -> bool {
+        let evm_addr2 = EvmAddress::from(0x1616060606060606060606060606060606060606060606060606060606060606);
+
+        evm_addr == evm_addr2
```

Comment:
Good idea. Better to return the comparison result rather than revert due to assert usage in sway. It helps narrow down the issue by separating it from a ton of other reasons why a revert might happen.

We should probably do it this way in the future, maybe even update all existing sway fn-s that use assertions to validate input args in some future general refactoring.

Reply:
Oh yeah that's smart